### PR TITLE
ADD A5/1 Algorithm

### PR DESCRIPTION
**Pull request by nirbhay12345**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #4977`.
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

**Comment by nirbhay12345** [comment]
@poyea please review this PR

File: ciphers/a51.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    freq = Counter(arr)
+    for (key,val) in freq.items():
+        if(val > 1):
+            return key
+
+def set_register_majority(majority_value:int) -> None:
+    """
+    >>> set_register_majority(0)
+    """
+    if (majority_value == LFSR1[8]):
```
Comment (poyea) [inline ciphers/a51.py:60]:
```suggestion
    if majority_value == LFSR1[8]:
```

File: ciphers/a51.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    if (mv == LFSR2[10]):
+        intermediate_xor = LFSR2[tapped_bits_1[0]] ^ LFSR2[tapped_bits_1[1]] 
+        LFSR2.insert(0, intermediate_xor)
+        LFSR2.pop()
+    if (mv == LFSR3[10]):
+        intermediate_xor = LFSR3[tapped_bits_1[0]] ^ LFSR3[tapped_bits_1[1]] ^ LFSR3[tapped_bits_1[2]] ^ LFSR3[tapped_bits_1[3]]
+        LFSR3.insert(0, intermediate_xor)
+        LFSR3.pop()
+
+def main() -> None:
```
[the remark this answers is not in the document]
Comment (poyea) [reply ciphers/a51.py:73]: We need more doctests

File: ciphers/a51.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+LFSR2:list = [] # 22 -> 10
+LFSR3:list = [] # 23 -> 10
+
+for _ in range(reg1_len):
+    LFSR1.append(0)
+for _ in range(reg2_len):
+    LFSR2.append(0)
+for _ in range(reg3_len):
+    LFSR3.append(0)
+
```
Comment (poyea) [inline ciphers/a51.py:19]: Use `[0] * N`?

Comment (nirbhay12345) [reply ciphers/a51.py:19]: okay working on this

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that do not have passing tests.

**Pull request closed without merging**